src/games/TicTacToe.ts

```diff
@@ -0,0 +1,248 @@
+import { fileBackedObject } from "../FileBackedObject";
+import Botty from "../Botty";
+
+import Discord = require("discord.js");
+import { stat } from "fs";
+
+interface TicTacToeScore {
+    lowWins: number;
+    highWins: number;
+    draws: number;
+}
+
+enum TicTacToeType {
+    X = "X", O = "O",
+}
+
+namespace TicTacToeType {
+    export function opposite(current: TicTacToeType): TicTacToeType | undefined {
+        if (current === TicTacToeType.X) return TicTacToeType.O;
+        if (current === TicTacToeType.O) return TicTacToeType.X;
+        return undefined;
+    }
+}
+
+interface TicTacToeGame {
+    board: { [key: number]: TicTacToeType };
+    turnPlayer: TicTacToeType;
+    Xplayer: string;
+    Oplayer: string;
+    scoreKey: string;
+    channel: Discord.TextChannel;
+}
+
+export default class TicTacToe {
+
+    private scores: { [key: string]: TicTacToeScore };
+    private games: TicTacToeGame[] = [];
+
+    constructor(client: Discord.Client, scorefile: string) {
+        this.scores = fileBackedObject(scorefile);
+        client.on("message", this.handleChat.bind(this));
+    }
+
+    public async onInvite(message: Discord.Message, isAdmin: boolean, command: string, args: string[]) {
+        const opponent = message.mentions.users.first();
+        if (!opponent) {
+            message.reply("You need to specify an opponent to start a TTT game!");
+            return;
+        }
+
+        const isBottyOpponent = opponent.bot;
+        if (isBottyOpponent) {
+            message.reply("Playing against Botty does not work at the moment.. :(");
+            return;
+        }
+
+        const senderName = message.author.username;
+        const opponentName = opponent.username;
+        const key = message.author.id.localeCompare(opponent.id) ? message.author.id + "_" + opponent.id : opponent.id + "_" + message.author.id;
+        const highP = message.author.id.localeCompare(opponent.id) ? senderName : opponentName;
+        const lowP = opponent.id.localeCompare(message.author.id) ? opponentName : senderName;
```

Comment:
These lines do not do what you think they do. localeCompare returns either a negative number, zero or a positive number. The function returns zero only if the values are the same, and since both positive and negative numbers are truthy, the first branch will always be taken (unless the player plays against themselves).